superset-frontend/plugins/plugin-chart-cartodiagram/src/components/ChartLayer.tsx

```diff
@@ -158,9 +179,13 @@ export class ChartLayer extends Layer {
 
       const chartComponent = createChartComponent(
         this.chartVizType,
-        feature,
+        // Seems like the registered chart components change some
+        // props in-place, which leads to unwanted side effects. Therefore,
+        // we only pass a copy.
+        JSON.parse(JSON.stringify(feature)),
```

Comment:
minor: maybe `lodash.clone(..)` could be used

https://www.geeksforgeeks.org/lodash-_-clone-method/